### ISSUE TITLE
When exporting the data the columns are not translated

### ISSUE THREAD

**Issue by Chardonneaur** [piwik/piwik#20830]
Hello,
When i export the data from Matomo, let's imagine that my interface is in French, then, the columns are not translated.
It would be nice to directly include the following parameter: &translateColumnNames=1
or to include it within the User Interface before we click on the export button.

**Comment by michalkleiner** [comment]
Hi @Chardonneaur,

it seems like when using CSV, TSV or RSS types of exports the translations should already be there. Have you tried those and it didn't work? Or you used different type of export where it's not supported (XML, JSON)?

If it was the first three and the translations didn't appear, could you provide replication steps and which report you exported etc? Thanks!

**Comment by sgiehl** [comment]
@Chardonneaur Could you answer @michalkleiner questions? Otherwise we may close this one, as it's not fully clear what exactly you are asking for.

**Comment by Chardonneaur** [comment]
@michalkleiner @sgiehl i tried in html and it does not work, https://demo.matomo.cloud/index.php?module=API&format=HTML&idSite=1&period=day&date=2023-09-10,2023-10-09&method=API.get&filter_limit=100&format_metrics=1&expanded=1&token_auth=anonymous&force_api_session=1

**Comment by sgiehl** [comment]
@Chardonneaur simply attaching `&translateColumnNames=1` to that url should provide translated names. So in general that works.

**Comment by Chardonneaur** [comment]
@sgiehl yes but for a normal human being, when clicking on the export button, while we are in the French UI, it should output translated columns without having to add a parameter within the url. That's the thing that I would like to stress. It is is a UX issue.

**Comment by sgiehl** [comment]
So you are looking for an option in the export overlay, that can be checked to enable/disable the column translation. right?

**Comment by Chardonneaur** [comment]
Well, it shouldn't even be an option. If I am using Matomo with a French User Interface, the export should show the columns translated in French and not in English.

**Comment by sgiehl** [comment]
We could maybe also automatically add that to HMTL exports as well. But for XML and JSON exports this won't be supported.
